README.md

```diff
@@ -1,3 +1,45 @@
-# PyWhyStats
+[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
+[![CircleCI](https://circleci.com/gh/py-why/pywhy-stats/tree/main.svg?style=svg)](https://circleci.com/gh/py-why/pywhy-stats/tree/main)
+[![unit-tests](https://github.com/py-why/pywhy-stats/actions/workflows/main.yml/badge.svg)](https://github.com/py-why/pywhy-stats/actions/workflows/main.yml)
+[![Checked with mypy](http://www.mypy-lang.org/static/mypy_badge.svg)](http://mypy-lang.org/)
+[![codecov](https://codecov.io/gh/py-why/pywhy-stats/branch/main/graph/badge.svg?token=H1reh7Qwf4)](https://codecov.io/gh/py-why/pywhy-stats)
 
-WIP
\ No newline at end of file
+# PyWhy-Stats
+
+Pywhy-stats serves as Python library for implementations of (un)conditional independence tests and related statistical functions, which can be utilized in various tasks, such as causal discovery.
+
+# Documentation
+
+See the [development version documentation](https://py-why.github.io/pywhy-stats/dev/index.html).
+
+Or see [stable version documentation](https://py-why.github.io/pywhy-stats/stable/index.html)
+
+# Installation
+
+Installation is best done via `pip` or `conda`. For developers, they can also install from source using `pip`. See [installation page](TBD) for full details.
+
+## Dependencies
+
+Minimally, pywhy-stats requires:
```

Comment:
Don't we also need scikit learn for the k-NN stuff and classifier/regression based methods?

Reply:
Since it's not necessary for most tests, we can just keep that optional for now since scikit-learn isn't necessary for things like Chi-square, FisherZ, etc. 

I kept sklearn in the `test` group poetry deps